### PR TITLE
move implementation of RustHermit's Mutex to hermit-abi

### PR DESCRIPTION
**Pull request by stlankes**
With this pull request, the OS-dependent implementation of the mutex is moved to `hermit-abi`. This should make `libstd` and `hermit-abi` easier to maintain.

### CODE REVIEW

File: library/std/src/sys/hermit/mutex.rs [hunk 1]
```diff
@@ -1,218 +1,33 @@
...
-    locked: bool,
-    blocked_task: PriorityQueue,
-}
-
-impl MutexInner {
-    pub const fn new() -> MutexInner {
-        MutexInner { locked: false, blocked_task: PriorityQueue::new() }
-    }
-}
+pub use crate::sys::hermit::abi;
```
Comment (bjorn3) [inline library/std/src/sys/hermit/mutex.rs:1]: Does this need to be `pub`?

Comment (stlankes) [reply library/std/src/sys/hermit/mutex.rs:1]: You are right, it isn't necessary and I will remove it.

File: library/std/Cargo.toml [hunk 2]
```diff
@@ -42,7 +42,7 @@ dlmalloc = { version = "0.2.3", features = ['rustc-dep-of-std'] }
 fortanix-sgx-abi = { version = "0.3.2", features = ['rustc-dep-of-std'] }
 
 [target.'cfg(target_os = "hermit")'.dependencies]
-hermit-abi = { version = "0.2.0", features = ['rustc-dep-of-std'] }
+hermit-abi = { version = "0.2.2", features = ['rustc-dep-of-std'] }
```
Comment (bjorn3) [inline library/std/Cargo.toml:45]: Please commit an updated `Cargo.lock` which reflects the bump to 0.2.2.

Comment (stlankes) [reply library/std/Cargo.toml:45]: Yes, I forgot to add the file

**Comment by stlankes** [comment]
I close this PR because @mkroening and I want to rethink the interface.

**Pull request closed without merging**